gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageReadChannel.java

```diff
@@ -811,7 +811,7 @@ private void cacheFooter(HttpResponse response) throws IOException {
       do {
         totalBytesRead += bytesRead;
         bytesRead = footerStream.read(footerContent, totalBytesRead, footerSize - totalBytesRead);
-      } while (bytesRead >= 0 && totalBytesRead <= footerSize);
+      } while (bytesRead >= 0 && totalBytesRead < footerSize);
```

Comment:
Why was this change done? Could someone explain more @medb  @mprashanthsagar Thanks

Reply:
We are seeing some infinite loop in our system in older versions while caching the footers.